src/daft-plan/src/planner.rs

```diff
@@ -103,23 +99,7 @@ pub fn plan(logical_plan: &LogicalPlan) -> DaftResult<PhysicalPlan> {
                     PartitionSpec::new(PartitionScheme::Unknown, mem_info.num_partitions, None)
                         .into(),
                 ));
-                let plan = if output_schema.fields.len() < mem_info.source_schema.fields.len() {
-                    let projection = output_schema
-                        .fields
-                        .iter()
-                        .map(|(name, _)| Expr::Column(name.clone().into()))
-                        .collect::<Vec<_>>();
-                    let partition_spec = scan.partition_spec().clone();
-                    PhysicalPlan::Project(Project::try_new(
-                        scan.into(),
-                        projection,
-                        Default::default(),
-                        partition_spec,
-                    )?)
-                } else {
-                    scan
-                };
-                Ok(plan)
+                Ok(scan)
```

Comment:
Is this safe? Don't we still need some mechanism to "prune" the columns in the output table?

Our InMemoryInfo doesn't contain any `Pushdowns` since it doesn't have a `ScanTask`, so how is it able to provide that pruning capabilities without an explicit `Projection`?

Reply:
We're no longer pruning projections that are downstream from in-memory scans, so we no longer need to manually insert back a column-selecting projection! 😄 

https://github.com/Eventual-Inc/Daft/blob/0c450a535065a7fe42f744c372898dcf45e8883b/src/daft-plan/src/optimization/rules/push_down_projection.rs#L178-L179

Reply:
Ah very meta... makes sense.